whitesands/code/modules/projectiles/guns/ballistic/automatic.dm

```diff
@@ -115,34 +113,37 @@
 	icon = 'icons/obj/guns/48x32guns.dmi'
 	fire_sound = 'sound/weapons/gun/rifle/shot.ogg'
 	icon_state = "ak47_nt"
-	item_state = "ntak"
+	item_state = "ak47_nt"
 	fire_rate = 5
 	mag_display = TRUE
 	weapon_weight = WEAPON_MEDIUM
 	w_class = WEIGHT_CLASS_BULKY
 	mag_type = /obj/item/ammo_box/magazine/aknt
 	var/folded = FALSE
 	var/unfolded_spread = 2
-	var/unfolded_recoil = 0
-	var/folded_spread = 7
-	var/folded_recoil = 0.50
+	var/unfolded_item_state = "ak47_nt"
+	var/folded_spread = 20
+	var/folded_item_state = "ak47_nt_stockless"
 
 /obj/item/gun/ballistic/automatic/ak47/nt/CtrlClick(mob/user)
 	. = ..()
 	if((!ishuman(user) || user.stat))
 		return
-	fold(user)
+	to_chat(user, "<span class='notice'>You start to [folded ? "unfold" : "fold"] the stock on the [src].</span>")
+	if(do_after(user, 10, target = src))
+		fold(user)
+		user.regenerate_icons()
```

Comment:
Find whichever update icon proc you need to call. Do not call regenerate icons, its expensive and gross overkill.

Reply:
this is fixed